include/global/state_space_data.hpp

```diff
@@ -191,46 +191,15 @@ namespace NP {
 					auto pred_idx = pred.first->get_job_index();
 
 					Interval<Time> ft{ 0, 0 };
-					s.get_finish_times(pred_idx, ft);
+					bool is_cert_finished;
+					s.get_finish_times(pred_idx, ft, is_cert_finished);
+
+					if (high_suspension.max() == 0 && is_cert_finished)
+						continue;
```

Comment:
I would push the `if (high_suspension.max() == 0 && is_cert_finished) continue` down into the `if (high_suspension.max() == 0) {` block a couple of lines later. This is just a minor preference of course.